## Logo del Tec

## Proyecto Integrador
## Smart Travel Assistant: Optimizando la Experiencia de Viaje con IA

### Equipo 37
##### A00759664 - Joel Orlando Hernández Ramos	
##### A01793486 - Juan Carlos Alvarado Carricarte
##### A00260430 - Juan Carlos Romo Cárdenas


**Cargando Conjunto de Datos**

In [1]:
# Importando librerias requeridas
import pyarrow.parquet as pq
import pandas as pd
import numpy as np


In [2]:
# Cargando conjunto de datos de hoteles como un Dataset multi-parte de Apache Parquet
dataset = pq.ParquetDataset('hotel-dataset')
table = dataset.read()

In [3]:
# Creando un Dataframe de Pandas para el analisis
dataframe = table.to_pandas()


**Estadisticas Resumidas del Conjunto de Datos**

El conjunto de datos de informacion de hoteles contiene mas de un millon novecientos tres mil registros con 14 columnas. Las columnas incluyen valores como la descripcion del hotel, el numero de estrellas del hotel, la direccion del hotel, el codigo del pais donde se localiza el hotel y sus coordenadas geograficas, entre otras.

Para crear la base de conocimientos en AWS Bedrock se esperan usar los siguientes campos o columnas:
* Nombre del hotel
* Descripción del hotel
* Nombre de la ciudad donde se ubica el hotel
* Código del país donde se ubica el hotel
* Nivel del hotel (rating)

Otros piezas de informacion que se pudieran usar incluyen:
* Longitud y Latitud


Uno de los parametros importantes para crear una base de conocimiento para RAG es el tamanio del texto a caracteres, or *chunck size*. Para este efecto se agregara una nueva columna con el tamanio de la descripcion del hotel en caracteres.



In [4]:
# Calculando el tamanio en caracteres de la descripcion del hotel
dataframe['DescLength'] = dataframe['Description'].str.len()


In [5]:
# Desplegando un resumen estadisticos de los datos
dataframe.describe(include='all')

HotelCode    HotelName  \
count    1093095      1093095   
unique   1093095       994885   
top      1688289  quality inn   
freq           1          373   
mean         NaN          NaN   
std          NaN          NaN   
min          NaN          NaN   
25%          NaN          NaN   
50%          NaN          NaN   
75%          NaN          NaN   
max          NaN          NaN   

                                              Description  \
count                                             1093095   
unique                                            1025518   
top     <br/><b>Disclaimer notification: Amenities are...   
freq                                                62806   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

                                           Address  Pincode CountryCode  \
count                                      1093095  1093095     1093095   
unique                                     1060951   178491         441   
top     510 gulf shore drive, , destin, 32541, usa                   US   
freq                                            94    40804      116324   
mean                                           NaN      NaN         NaN   
std                                            NaN      NaN         NaN   
min                                            NaN      NaN         NaN   
25%                                            NaN      NaN         NaN   
50%                                            NaN      NaN         NaN   
75%                                            NaN      NaN         NaN   
max                                            NaN      NaN         NaN   

           PhoneNumber CityName  Longitude Latitude   HotelRating  \
count           742223  1093095    1090879  1093095  1.093052e+06   
unique          612681   100184    1009164   980655           NaN   
top     91-93-13931393     Rome  -86.49782                    NaN   
freq              6690     7937         94     2217           NaN   
mean               NaN      NaN        NaN      NaN  4.772850e+01   
std                NaN      NaN        NaN      NaN  4.784483e+04   
min                NaN      NaN        NaN      NaN  0.000000e+00   
25%                NaN      NaN        NaN      NaN  0.000000e+00   
50%                NaN      NaN        NaN      NaN  2.000000e+00   
75%                NaN      NaN        NaN      NaN  3.000000e+00   
max                NaN      NaN        NaN      NaN  5.002136e+07   

                                        uuid      match_id  \
count                                1093095  1.093095e+06   
unique                               1093095           NaN   
top     e8558524-a7e1-4c2c-a4aa-afaa8adc2563           NaN   
freq                                       1           NaN   
mean                                     NaN  2.839610e+12   
std                                      NaN  1.744277e+12   
min                                      NaN  0.000000e+00   
25%                                      NaN  1.322850e+12   
50%                                      NaN  2.843268e+12   
75%                                      NaN  4.355097e+12   
max                                      NaN  5.858335e+12   

        match_confidence_score    DescLength  
count                  24688.0  1.093095e+06  
unique                     NaN           NaN  
top                        NaN           NaN  
freq                       NaN           NaN  
mean                       1.0  1.356845e+03  
std                        0.0  5.874179e+02  
min                        1.0  0.000000e+00  
25%                       

La descripcion extendida de los datos almacenados en en conjunto de datos indica que:
* Hay 1,093,095 registros con 14 columnas
* Las columnas *PhoneNumber* y *Longitude* tienen valores ausentes
* El nivel promedio de estrellas de los hoteles es por encima de 4 estrellas de acuerdo a la media de *HotelRating*
* El tamanio promedio en caracteres de las descripcion de un hotel es 1,356 de acuerdo a la media de *DescLength*, aunque hay descripciones vacias basados en el valor minimo de cero (0) de la misma columna


In [6]:
# Desplegando los tipos de valor almacenados en las diferentes columnas con contador de valores no ausentes
dataframe.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1093095 entries, 0 to 1093094
Data columns (total 15 columns):
 #   Column                  Non-Null Count    Dtype  
---  ------                  --------------    -----  
 0   HotelCode               1093095 non-null  object 
 1   HotelName               1093095 non-null  object 
 2   Description             1093095 non-null  object 
 3   Address                 1093095 non-null  object 
 4   Pincode                 1093095 non-null  object 
 5   CountryCode             1093095 non-null  object 
 6   PhoneNumber             742223 non-null   object 
 7   CityName                1093095 non-null  object 
 8   Longitude               1090879 non-null  object 
 9   Latitude                1093095 non-null  object 
 10  HotelRating             1093052 non-null  float64
 11  uuid                    1093095 non-null  object 
 12  match_id                1093095 non-null  int64  
 13  match_confidence_score  24688 non-null    float64
 14  De

El desplegado de la informacion del conjunto de datos tambien confirma la cuenta de registros totales y de registros ausentes indicados anteriormente.

In [7]:
# Desplegando una muestra de los valores presentes en el conjunto de datos
dataframe.head()

HotelCode                HotelName  \
0   1688289               villa alun   
1   1863020          house of ahasna   
2   5414820  casa vacanze margherita   
3   5758326              casa bethel   
4   1116957              pension abc   

                                         Description  \
0  <p>HeadLine : In Seminyak (Batubelig)</p><p>Lo...   
1  Located in Katunayaka, House Of Ahasna feature...   
2  Casa Vacanze Margherita is a detached holiday ...   
3  Located in Cobán, in a building dating from 20...   
4  This quietly located guest house in Berlin off...   

                                             Address Pincode CountryCode  \
0  jalan raya batu belig gang kamboja no.2, gang ...   80361          ID   
1  air force road, kuranakatunayake 109akatunayak...   11450          lk   
2         localita' piricone, , orosei, 08028, italy   08028          IT   
3  9na avenida 2 07 zona 1 coban, , coban, 16001,...   16001          GT   
4  kurfürstenstr. 20, 10785 berlinschöneberg, ber...   10785          DE   

       PhoneNumber    CityName   Longitude   Latitude  HotelRating  \
0  62-812-38144235    Seminyak  115.148258  -8.672673          4.0   
1             None  Katunayaka    79.86482    7.18265          3.0   
2             None      Orosei      9.6809   40.36094          0.0   
3             None       Coban   -90.37927   15.46935          0.0   
4  (49) 3026949903      Berlin    13.36543   52.49947          3.0   

                                   uuid       match_id  \
0  e8558524-a7e1-4c2c-a4aa-afaa8adc2563  1005022348356   
1  fe5a67fb-7140-43fc-a62c-b2521f86f257  4741643896104   
2  86d185f6-1e7b-4a24-8ba2-18e8b0edcf1b   927712937388   
3  b93537ec-e148-42b0-9bd9-2347b765f611  5592047420856   
4  ee2eb159-87a7-4b7b-8b2a-fbcb8d92dcf6   137438955021   

   match_confidence_score  DescLength  
0                     NaN        2325  
1                     NaN         732  
2                     NaN         898  
3                     NaN         480  
4                     NaN        1662

Una revision preliminar de los datos muestra lo siguiente:
* La descripcion del hotel incluye elementos, o tags, de HTML que se tienen que remover
* El codigo del pais donde reside el hotel parece estar basado en el estandard ISO 3166 (ISO, s.f.) de dos letras. Este es un data categorico con una cardinalidad de 249, ya que el estandar ISO 3166 incluye 249 codigos.


**Analisis de Datos Faltantes**



In [9]:
# Determinando columnas con valores ausentes
dataframe.isnull().sum()

HotelCode                       0
HotelName                       0
Description                     0
Address                         0
Pincode                         0
CountryCode                     0
PhoneNumber                350872
CityName                        0
Longitude                    2216
Latitude                        0
HotelRating                    43
uuid                            0
match_id                        0
match_confidence_score    1068407
DescLength                      0
dtype: int64

El analisis preliminar de datos faltantes muestra lo siguiente:
* Aproximadamente tercera parte de los registros, mas de 350 mil, no tiene numeros telefonicos
* Aunque faltan mas de dos mil doscientos valore de logitud, no faltan valors de latitud

A continuacion se hara un analisis de datos categoricos y numericos unicamente, por lo que se creara un conjunto de datos que excluya valores no requeridos.

In [11]:
# Removiendo campos con texto para analisis estadistico
dataframe_analysis = dataframe.drop(['Description','PhoneNumber','CityName','Address','Pincode','uuid','match_id','match_confidence_score'], axis=1)

In [12]:
dataframe_analysis.describe(include='all')

HotelCode    HotelName CountryCode  Longitude Latitude   HotelRating  \
count    1093095      1093095     1093095    1090879  1093095  1.093052e+06   
unique   1093095       994885         441    1009164   980655           NaN   
top      1688289  quality inn          US  -86.49782                    NaN   
freq           1          373      116324         94     2217           NaN   
mean         NaN          NaN         NaN        NaN      NaN  4.772850e+01   
std          NaN          NaN         NaN        NaN      NaN  4.784483e+04   
min          NaN          NaN         NaN        NaN      NaN  0.000000e+00   
25%          NaN          NaN         NaN        NaN      NaN  0.000000e+00   
50%          NaN          NaN         NaN        NaN      NaN  2.000000e+00   
75%          NaN          NaN         NaN        NaN      NaN  3.000000e+00   
max          NaN          NaN         NaN        NaN      NaN  5.002136e+07   

          DescLength  
count   1.093095e+06  
unique           NaN  
top              NaN  
freq             NaN  
mean    1.356845e+03  
std     5.874179e+02  
min     0.000000e+00  
25%     9.770000e+02  
50%     1.383000e+03  
75%     1.750000e+03  
max     1.959800e+04

In [13]:
dataframe_analysis.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1093095 entries, 0 to 1093094
Data columns (total 7 columns):
 #   Column       Non-Null Count    Dtype  
---  ------       --------------    -----  
 0   HotelCode    1093095 non-null  object 
 1   HotelName    1093095 non-null  object 
 2   CountryCode  1093095 non-null  object 
 3   Longitude    1090879 non-null  object 
 4   Latitude     1093095 non-null  object 
 5   HotelRating  1093052 non-null  float64
 6   DescLength   1093095 non-null  int64  
dtypes: float64(1), int64(1), object(5)
memory usage: 58.4+ MB


In [14]:
dataframe_analysis.head()

HotelCode                HotelName CountryCode   Longitude   Latitude  \
0   1688289               villa alun          ID  115.148258  -8.672673   
1   1863020          house of ahasna          lk    79.86482    7.18265   
2   5414820  casa vacanze margherita          IT      9.6809   40.36094   
3   5758326              casa bethel          GT   -90.37927   15.46935   
4   1116957              pension abc          DE    13.36543   52.49947   

   HotelRating  DescLength  
0          4.0        2325  
1          3.0         732  
2          0.0         898  
3          0.0         480  
4          3.0        1662

La descripcion extendida del conjunto de analysis, asi como su informacion desplegada concuerdan con los valores obtenidos del conjunto de datos original.

Como las coordenadas geograficas de longitud y latitud son candidatas para agregarse a la base de conocimiento es pertinente revisar los valores ausentes para ver si hay forma de completarlos.

In [16]:
dataframe.loc[dataframe['Longitude'].isnull()]

HotelCode                                    HotelName  \
352       1944494                   sunview lodge & restaurant   
429       5001876      santubong suites b just like home damai   
1432      5002452                                 hotel salyut   
2031      5020924         atour hotel shanghai pudong lujiazui   
3066      5016410        premier inn bangor (northern ireland)   
...           ...                                          ...   
1091396   5015364                            iris lite corbett   
1092511   5007324        jinjiang innselect  xinhua road wuhan   
1092668   5010802  ji hotel shanghai wujiaochang shiguang road   
1092679   5021993                      bristol hotel podgorica   
1093093   1298444                             hotel de leurope   

                                               Description  \
352      <br/><b>Disclaimer notification: Amenities are...   
429      <p>Featuring an outdoor swimming pool, a fitne...   
1432     <p>The hotel "salute" was founded in 2007. It ...   
2031     <p> This hotel offers a pleasant stay in Shang...   
3066     Whether you’re planning romantic breaks or fam...   
...                                                    ...   
1091396  Our core focus is to provide a budget friendly...   
1092511  <p>While staying at Jinjiang Inn (Wuhan Xinhua...   
1092668  <p> JI Hotel Shanghai Wujiaochang Shiguang Roa...   
1092679  <p>Situated in Podgorica, 700 metres from Chur...   
1093093  &lt;p&gt;&lt;b&gt;Property Location&lt;/b&gt; ...   

                                                   Address Pincode  \
352      mombasa road, kibwezi 90137, kibwezi, 90137, k...   90137   
429      jalan sultan tengah, 93050 kuching, malaysia, ...   93050   
1432     tula region, city aleksin, 18, bolotova st., ,...           
2031     no. 138 pudong avenue lujiazui, pudong 200000 ...           
3066     castle avenue, bangor, northern ireland bt20 4...           
...                                                    ...     ...   
1091396       dhikuli village,, ramnagar, ramnagar, 244715  244715   
1092511        no. 162 north jianghan, roadwuhan, nanjing,           
1092668  no.635 shiguang road, shanghai, pvg, 0000, sha...           
1092679  2 bore stankovica,, 81000 podgorica, montenegr...   81000   
1093093  29 avenue de royat, 63400 chamalieres, chamali...   63400   

        CountryCode PhoneNumber     CityName Longitude Latitude  HotelRating  \
352              KE        None      Kibwezi      None                   0.0   
429              MY        None  Damai Beach      None                   4.0   
1432             RU        None      Aleksin      None                   3.0   
2031             CN        None     Shanghai      None                   4.0   
3066             GB        None       Bangor      None                   0.0   
...             ...         ...          ...       ...      ...          ...   
1091396          13  6398601206     Ramnagar      None                   3.0   
1092511          CN        None      NANJING      None                   0.0   
1092668          CN        None     Shanghai      None                   0.0   
1092679          ME        None    Podgorica      None                   0.0   
1093093          FR        None  Chamalières      None                   3.0   

                                         uuid       match_id  \
352      1b539168-5527-45a1-a3e5-dfcdba5c0f90  1760936592369   
429      d51aa7de-79f8-4ea0-831a-6607a6fdadd3  5643587027080   
1432     9c8143f7-b995-4968-a82f-77258df3ca4b  5643587027641   
2031     a129f8e4-4275-460d-9b47-bb1723a781fc  3341484557220   
3066     956944a4-1d0b-4024-b903-90465cd86463  4294967296355   
...                                       ...            ...   
1091396  b359cb1b-91ad-44ba-8e35-e6731d02659e  4724464026634   
1092511  92d0a2c3-be2d-4284-97d3-4457138e9a37  4698694222010   
1092668  58917d5c-df4c-4292-bcad-140b71977686  2319282340207   
1092679  a7d30a6e-a847-4fdc-

Un hallazgo importante del desplegado anterior es que, en los registros donde el valor de longitud no esta presente. la latitud no presenta un valor aun y cuando el analisis de valores ausentes indica que cuenta con valores para todos los registros.

Para poder verificar este hallazgo se convertiran los valores de longitud y latitud a un valor de punto flotante valido o faltante.

In [17]:
# Convirtiendo latitud a un numero
def to_float_number(x):
    try:
        return float(x)
    except:
        return np.nan

dataframe_analysis['Longitude'] = dataframe_analysis['Longitude'].apply(to_float_number) 
dataframe_analysis['Latitude'] = dataframe_analysis['Latitude'].apply(to_float_number)

In [18]:
dataframe_analysis.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1093095 entries, 0 to 1093094
Data columns (total 7 columns):
 #   Column       Non-Null Count    Dtype  
---  ------       --------------    -----  
 0   HotelCode    1093095 non-null  object 
 1   HotelName    1093095 non-null  object 
 2   CountryCode  1093095 non-null  object 
 3   Longitude    1066303 non-null  float64
 4   Latitude     1066296 non-null  float64
 5   HotelRating  1093052 non-null  float64
 6   DescLength   1093095 non-null  int64  
dtypes: float64(3), int64(1), object(3)
memory usage: 58.4+ MB


In [19]:
# Determinando columnas con valores ausentes
dataframe_analysis.isnull().sum()

HotelCode          0
HotelName          0
CountryCode        0
Longitude      26792
Latitude       26799
HotelRating       43
DescLength         0
dtype: int64

Despues de la conversion se puede observar que logitud y latitud tienen una cuenta de valores ausentes similar, aunque no la misma.


In [59]:
dataframe_desc = dataframe.loc[dataframe['DescLength']<150].drop(['PhoneNumber','CityName','Address','Pincode','uuid','match_id','match_confidence_score'], axis=1)


In [60]:
dataframe_desc.info()

<class 'pandas.core.frame.DataFrame'>
Index: 63228 entries, 29 to 1093088
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   HotelCode    63228 non-null  object 
 1   HotelName    63228 non-null  object 
 2   Description  63228 non-null  object 
 3   CountryCode  63228 non-null  object 
 4   Longitude    62959 non-null  object 
 5   Latitude     63228 non-null  object 
 6   HotelRating  63227 non-null  float64
 7   DescLength   63228 non-null  int64  
dtypes: float64(1), int64(1), object(6)
memory usage: 4.3+ MB


In [61]:
dataframe_desc.head(40)

HotelCode                       HotelName  \
29    5574247                   artinov villa   
42    5227047                cabanas wualmapu   
69    1100700               lewis grand hotel   
120   1397984                   nour el balad   
124   5617885             rigofutty vendeghaz   
165   5474897         pokoje goscinne koralik   
177   5301104                 apartment cesar   
184   5883785              casa baixú caraíva   
185   5441589                       el vergel   
196   6186660                    bella pensao   
213   1672944  villas y bungalows tlaquepaque   
242   1512518                  monalisa lodge   
268   5910033               ella peace heaven   
288   1623255        szeker fogado es etterem   
289   1325169                    yamato hotel   
314   1761438                     nash hostel   
352   1944494      sunview lodge & restaurant   
384   5929071              chamee's home stay   
388   1528846             solanki guest house   
404   6012371                             mir   
414   5993829                 hosteria wikter   
446   1001752            künszler apartmanház   
449   1707601     indaba lodge hotel gaborone   
456   1923925                  hostal quentin   
470   1776390                  casal dei papi   
473   5156293                    roberto vila   
508   6118717                       101norfol   
511   5183996          apartments palma rabac   
526   5689967             el studio apartment   
532   5168077                 zielone wzgórze   
536   5304466                 villa harzblick   
546   5277029                cazare la troita   
584   5576630             residencial do tchê   
588   5894695            green tara home stay   
603   5661610                      casa sahar   
627   1469022                  gaga bees yala   
642   5993522                     strandhafer   
645   5900071               pravasi residency   
649   6192237               peacefuldalhousie   
788   1770719                    hotel silver   

                                           Description CountryCode  \
29   <br/><b>Disclaimer notification: Amenities are...          UA   
42   <br/><b>Disclaimer notification: Amenities are...          CL   
69   <br/><b>Disclaimer notification: Amenities are...          PH   
120  <br/><b>Disclaimer notification: Amenities are...          EG   
124  <br/><b>Disclaimer notification: Amenities are...          HU   
165  <br/><b>Disclaimer notification: Amenities are...          PL   
177  <br/><b>Disclaimer notification: Amenities are...          SI   
184  <br/><b>Disclaimer notification: Amenities are...          BR   
185  <br/><b>Disclaimer notification: Amenities are...          AR   
196  <br/><b>Disclaimer notification: Amenities are...          IN   
213  <br/><b>Disclaimer notification: Amenities are...          MX   
242  <br/><b>Disclaimer notification: Amenities are...          IN   
268  <br/><b>Disclaimer notification: Amenities are...          LK   
288  <br/><b>Disclaimer notification: Amenities are...          HU   
289  <br/><b>Disclaimer notification: Amenities are...          TH   
314  <br/><b>Disclaimer notification: Amenities are...          RU   
352  <br/><b>Disclaimer notification: Amenities are...          KE   
384  <br/><b>Disclaimer notification: Amenities are...          LK   
388  <br/><b>Disclaimer notification: Amenities are...          IN   
404  <br/><b>Disclaimer notification: Amenities are...          HR   
414  <br/><b>Disclaimer notification: Amenities are...          AR   
446  <br/><b>Disclaimer notification: Amenities are...          HU   
449  <br/><b>Disclaimer notification: Amenities are...          BW   
456  <br/><b>Disclaimer notification: Amenities are...          EC   
470  <br/><b>Disclaimer notification: Amenities are...          IT   
473  <br/><b>Disclaimer notification: Amenities are...          LT   
508  <br/><b>Disclaimer notification: Amenities are...          US   
511  <br/><b>Disclaimer notificati

In [62]:
dataframe_desc = dataframe_desc.loc[dataframe_desc['DescLength']<100]

In [63]:
dataframe_desc.info()

<class 'pandas.core.frame.DataFrame'>
Index: 94 entries, 26256 to 1037114
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   HotelCode    94 non-null     object 
 1   HotelName    94 non-null     object 
 2   Description  94 non-null     object 
 3   CountryCode  94 non-null     object 
 4   Longitude    94 non-null     object 
 5   Latitude     94 non-null     object 
 6   HotelRating  93 non-null     float64
 7   DescLength   94 non-null     int64  
dtypes: float64(1), int64(1), object(6)
memory usage: 6.6+ KB


In [64]:
dataframe_desc.head(40)

HotelCode                                          HotelName  \
26256        4dWK                        tesoro ixtapa all inclusive   
36715        tJ67               americinn by wyndham wisconsin dells   
48092        kD6w                  camino real hotel & suites puebla   
51347        L0F5                     capital o hotel posada terraza   
53828        R4fM     homewood suites by hilton austin round rock tx   
62935        kk3L      ramada by wyndham viscount suites tucson east   
69356        PfXB                                   camelback resort   
82836   HotelCode                                          hotelname   
121147       9fqD                    quality inn litchfield route 66   
125927       XFhS            disney's all star movies resort package   
158038       SmHs   country inn & suites by radisson, orangeburg, sc   
158925       mvPA                          courtyard upper east side   
165995       Uv8I             americinn lodge & suites lincoln south   
172705       8Khm                     baymont by wyndham west plains   
173338       jWWw                    hilton orange county/costa mesa   
176657       37P1   country inn & suites by radisson, rock falls, il   
196021       wpnU                        wyndham hotel orange county   
213561       Wymh  grand legacy at the park (formerly ramada main...   
222412       Dqvs     delta hotels by marriott detroit metro airport   
222765       usZ3                                      hotel windrow   
244868       y482                best western premier ivy hotel napa   
248925       4yYv  country inn & suites by radisson, port clinton...   
253728       zjmd                              smart cancun by oasis   
269118       93wY  wingate by wyndham los angeles international a...   
285126       Ryxs       best western plus flint airport inn & suites   
292920       uZNv  hampton inn indianapolis - downtown circle centre   
297061       PKzv       best western bloomington edina - minneapolis   
319719       B9td                     real inn cancun by camino real   
332004       ShXC  country inn & suites by radisson, atlanta i-75...   
355776       FSbv  country inn & suites by radisson, madison west...   
358187       dr24  homewood suites by hilton rancho cordova sacra...   
366825       7nzQ                americas best value inn sauk centre   
421322       rVhu                                       oasis cancun   
426165       V31y  howard johnson hotel&conf cntr by wyndham full...   
429955       8PKt                                  hotel seven crown   
470332       jNZH  city express by marriott monterrey santa catarina   
477544       5dDR                            red roof inn clearfield   
484728       zCJe         tesoro los cabos - all inclusive available   
485763       gZaH                                     the view hotel   
495157       EBMl  holiday inn express & suites ann arbor - unive...   

                                              Description  CountryCode  \
26256   <p><b>About the property</b><br /><span></span...           MX   
36715    This cosy hotel is set in Wisconsin Rapids Area.           US   
48092                                   Rooms Number: 149           MX   
51347                                                               MX   
53828                                                               US   
62935   This simple hotel is located in Downtown/Unive...           US   
69356                                                               US   
82836                                         Description  CountryCode   
121147  Easy interstate access, riverboat gambling 50 ...           US   
125927                                                              US   
158038  This unpretentious hotel is situated in Orange...           US   
158925                                                              US   
165995  <p><b>About the property</b><br /><span> </spa...           US   
172705        This 

**Referencias**

* ISO. (s.f.) ISO 3166 — Country Codes. Internation Standards Organization. Recuperado 03 de mayo, 2024 de https://www.iso.org/iso-3166-country-codes.html
